src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py

```diff
@@ -0,0 +1,112 @@
+import pathlib
+
+from mantid.api import AlgorithmFactory, MatrixWorkspaceProperty, PropertyMode, PythonAlgorithm
+from mantid.kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "LoadGroupingDefinition"
+
+
+class LoadGroupingDefinition(PythonAlgorithm):
+    """
+    This algorithm creates a grouping workspace from a grouping definition file.
+    inputs:
+        GroupingFilename: str -- path of an input grouping definition file (NEXUS, XML, or DiffCal/HDF5)
+        InstrumentName: str -- name of an associated instrument
+        InstrumentFilename: str -- path of an associated instrument definition file
+        InstrumentDonor: str -- Workspace to optionally take the associate instrument from
+        OutputWorkspace: str -- name of an output grouping workspace
+    """
+
+    def PyInit(self) -> None:
+        # declare properties
+        self.declareProperty(
+            "GroupingFilename",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="Path of an input grouping definition file",
+        )
+        self.declareProperty(
+            "InstrumentName", defaultValue="", direction=Direction.Input, doc="Name of an associated instrument"
```

Comment:
trailing comma at end

Reply:
Change to 
``` python
self.declareProperty(
     "InstrumentName", 
    defaultValue="", 
    direction=Direction.Input, 
    doc="Name of an associated instrument",
)
```
Without the trailing comma, the linter will flatten it to one line.

Reply:
Sure.  I was just waiting for more requests to bunch them all up in one commit :)